### PR TITLE
🧪 [Testing Improvement] Edge case test and fix in AbstractReactiveFeignConfigurator

### PR DESCRIPTION
**Pull request by Periecle**
🎯 **What:** Missing edge case test in AbstractReactiveFeignConfigurator and fix for a bug in the `compareTo` method where comparing an object to itself would throw an `IllegalArgumentException`.
📊 **Coverage:** Added unit tests in `AbstractReactiveFeignConfiguratorTest` covering same instance comparison, different instance with same order comparison (exception expected), and different order comparison.
✨ **Result:** Improved test coverage and fixed a bug that violated the `Comparable` contract.

---
*PR created automatically by Jules for task [14191844338616150843](https://jules.google.com/task/14191844338616150843) started by @Periecle*

### CODE REVIEW

File: feign-reactor-spring-configuration/src/test/java/reactivefeign/spring/config/AbstractReactiveFeignConfiguratorTest.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    @Test
+    public void shouldThrowExceptionWhenSameOrder() {
+        TestConfigurator configurator1 = new TestConfigurator(1);
+        TestConfigurator configurator2 = new TestConfigurator(1);
+
+        assertThatThrownBy(() -> configurator1.compareTo(configurator2))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining("Same order for different configurators");
+    }
+
```
[the remark this answers is not in the document]
Comment (Periecle) [reply feign-reactor-spring-configuration/src/test/java/reactivefeign/spring/config/AbstractReactiveFeignConfiguratorTest.java:35]: @jules The tests cover reflexivity and anti-symmetry from the Comparable contract, which is great. To make the test suite more comprehensive, I suggest also adding a test for transitivity ((x.compareTo(y) > 0 && y.compareTo(z) > 0) implies x.compareTo(z) > 0). This will ensure all key properties of the contract are verified.

    @Test
    public void shouldBeTransitive() {
        TestConfigurator configurator1 = new TestConfigurator(1);
        TestConfigurator configurator2 = new TestConfigurator(2);
        TestConfigurator configurator3 = new TestConfigurator(3);

        // Verifies that if c1 < c2 and c2 < c3, then c1 < c3
        assertThat(configurator1.compareTo(configurator2)).isLessThan(0);
        assertThat(configurator2.compareTo(configurator3)).isLessThan(0);
        assertThat(configurator1.compareTo(configurator3)).isLessThan(0);
    }
